brmp/utils.py

```diff
@@ -17,3 +19,36 @@ def flatten(arr):
 # (num_chains * num_samples, ...) -> (num_chains, num_samples, ...)
 def unflatten(arr, num_chains, num_samples):
     return arr.reshape((num_chains, num_samples) + arr.shape[1:])
+
+
+class traceback_generated(object):
+    """
+    Context manager for augmenting exception message when running generated
+    code. The particular line from the generated code that resulted in the
+    exception is highlighted. This has no effect if the error originated
+    from elsewhere in the code.
+
+    :param str code: String representation of the code being run.
+    :raises: ModelSpecificationError
+    """
+    def __init__(self, code):
+        self.code = code
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        tb_info = traceback.extract_tb(exc_tb)
+        filename, line, fn, _ = tb_info[-1]
```

Comment:
I think this throws an error when no exception is raised in the model, since `tb_info==[]`. (CI passed because we're currently skipping tests, but tests fail locally.)

Reply:
Good point. Let me add some checks, and update the PR.